blockchain/blockchain.go

```diff
@@ -1116,7 +1118,7 @@ func (b *Blockchain) saveNodeAsChildNode(parentHash, childHash []byte) error {
 		return fmt.Errorf("failed to insert child node item under parent node: %w", err)
 	}
 
-	err = b.db.Put([]byte(channelsNodesCountPrefix), itemsUint64)
+	err = b.db.Put(append([]byte(channelsNodesCountPrefix), parentHash...), itemsUint64)
```

Comment:
I believe this is an issue, it will save an additional row for each different parenHash, correct?

Reply:
Yes this is the idea, for each channel item node, we keep track of its child since we want to skip iterating over keys to find the total count because its an expensive operation so we keep a separate record. Note that goleveldb is not a relational db where you can just do a count and get the total. This is the best way to implement a count

Reply:
Good explanation